server/controllers/AuthController.js

```diff
@@ -74,13 +77,111 @@ class AuthController {
         });
       }
       req.user = AuthController.stripeUser(newUser);
-      return res.status(201).json({
-        status: 'success',
-        user: AuthController.stripeUser(newUser),
-        message: 'User signup successful',
-      });
+      req.emailVerificationMessage = 'User signup successful and verification email sent.';
+      next();
     });
   }
+
+  /**
+   * Sends an email verification email to the user with a verification url
+   * @param {object} req the request object
+   * @param {object} res the response object
+   * @returns null
+   */
+  static verifyEmail(req, res, next) {
+    const { email } = req.user || req.body;
+    const payload = { email };
+    const token = JwtHelper.createToken(payload, '24h');
+    const url = `${req.protocol}://${req.headers.host}/api/auth/verify?evc=${token}`;
+    const msg = emails.emailVerification(email, url);
+    Mailer.sendMail(msg)
+      .then((response) => {
+        if (response[0].statusCode === 202) {
+          return res.status(201).json({
+            status: 'success',
+            message: req.emailVerificationMessage,
+            url,
+            token,
+          });
+        }
+        res.status(400).send('Unable to send email');
+      })
+      .catch(err => next(err, req, res));
+  }
+
+  /**
+   * Verifies a user account
+   * @param {object} req the request object
+   * @param {object} res the response object
+   */
+  static activateUser(req, res, next) {
+    const token = req.query.evc;
+    if (token) {
```

Comment:
Nice work @madeofhuman, I appreciate the fact that you took out your time to get this task done. I love the email template format you used for this. However, I feel you missed something crucial on line 119-124 of AuthController, you only proceeded with the request when the token is in the query parameter, then you have a nexted if which only proceeds when the token is still valid. I feel you feel you should have a fallback to return to the user when a token is not provided or is not valid. This would enable the user to know what is wrong and also ensures that a request is not stalled in its lifecycle. Thank you for completing this feature.